applications/pfem_2_application/custom_utilities/particle_utilities.h

```diff
@@ -638,127 +554,168 @@ namespace Kratos
 		iii->GetGeometry()[3].FastGetSolutionStepValue(NORMAL) += area_normal/ norm_c;
 		
 	      }
-	    if( iii->GetGeometry()[0].FastGetSolutionStepValue(IS_FREE_SURFACE) == 1.0 && iii->GetGeometry()[3].FastGetSolutionStepValue(IS_FREE_SURFACE) == 1.0 && iii->GetGeometry()[2].FastGetSolutionStepValue(IS_FREE_SURFACE) == 1.0)
-	      {
-		v1[0] =  iii->GetGeometry()[0].X() -iii->GetGeometry()[2].X();
-		v1[1] = iii->GetGeometry()[0].Y() - iii->GetGeometry()[2].Y();
-		v1[2] = iii->GetGeometry()[0].Z() - iii->GetGeometry()[2].Z();
+	      if( iii->GetGeometry()[0].FastGetSolutionStepValue(IS_BOUNDARY) == 1.0 && iii->GetGeometry()[3].FastGetSolutionStepValue(IS_BOUNDARY) == 1.0 && iii->GetGeometry()[2].FastGetSolutionStepValue(IS_BOUNDARY) == 1.0)
+		{
+		  v1[0] =  iii->GetGeometry()[0].X() -iii->GetGeometry()[2].X();
+		  v1[1] = iii->GetGeometry()[0].Y() - iii->GetGeometry()[2].Y();
+		  v1[2] = iii->GetGeometry()[0].Z() - iii->GetGeometry()[2].Z();
 		
-		v2[0] = iii->GetGeometry()[3].X() - iii->GetGeometry()[2].X();
-		v2[1] = iii->GetGeometry()[3].Y() - iii->GetGeometry()[2].Y();
-		v2[2] = iii->GetGeometry()[3].Z() - iii->GetGeometry()[2].Z();
-		MathUtils<double>::CrossProduct(area_normal,v1,v2);
-		//area_normal *= -0.5;
-		array_1d<double,3> msAuxVec = ZeroVector(3);
-		double c0 = fabs(area_normal[0]);
-		double c1 = fabs(area_normal[1]);
-		double c2 = fabs(area_normal[2]);
-		msAuxVec[0]=c0;
-		msAuxVec[1]=c1;
-		msAuxVec[2]=c2;
-		//double norm_c =norm_2(msAuxVec);
+		  v2[0] = iii->GetGeometry()[3].X() - iii->GetGeometry()[2].X();
+		  v2[1] = iii->GetGeometry()[3].Y() - iii->GetGeometry()[2].Y();
+		  v2[2] = iii->GetGeometry()[3].Z() - iii->GetGeometry()[2].Z();
+		  MathUtils<double>::CrossProduct(area_normal,v1,v2);
+		  //area_normal *= -0.5;
+		  array_1d<double,3> msAuxVec = ZeroVector(3);
+		  double c0 = fabs(area_normal[0]);
```

Comment:
this is ok. just as a comment, you could use alternatively std::abs() which does the correct overload